### ISSUE TITLE
Screenshot Listener failed to listen on android sdk 29

### ISSUE THREAD

**Issue by dopefaceee** [flutter-moum/flutter_screenshot_callback#29]
Hi, i tried this plugin on android sdk version 29. and seems like it's not working with addListener. 
Working well with sdk 28. still not sure why this happened.

**Comment by dariuspo** [comment]
i have same issue, with this.
copied whole example.dart the addListener is not firing

**Comment by giovannilima262** [comment]
I have same issue.

**Comment by beygee** [comment]
@dopefaceee 

Thank you for opening the issue.

I tested my app with sdk 29. That test went well.

This plugin uses external storage permission. The permission must be approved.

May I ask you to check if the permission has been approved?

**Comment by RohithKilaparthi** [comment]
@beygee I tested it on the physical devices and it doesn't work

**Comment by siloebb** [comment]
```
D/SmartClipDataCropperImpl(29541): sendExtractionResultToSmartClipService : -- Extracted SmartClip data information --
D/SmartClipDataCropperImpl(29541): sendExtractionResultToSmartClipService : Request Id : 22
D/SmartClipDataCropperImpl(29541): sendExtractionResultToSmartClipService : Extraction mode : 1
D/SemSmartClipDataRepository(29541): ----- Start of SmartClip repository informations -----
D/SemSmartClipDataRepository(29541): ** Content type : image
D/SemSmartClipDataRepository(29541): ** Meta area rect : Rect(0, 0 - 0, 0)
D/SemSmartClipDataRepository(29541): ** Captured image file path : null
D/SemSmartClipDataRepository(29541): ----- End of SmartClip repository informations -----
D/SmartClipDataCropperImpl(29541): sendExtractionResultToSmartClipService : Elapsed = 2
```
I get this error in my Android when I try to take a screenshot.  I just trying to print a message in the console.

```
Future<void> initScreenshotCallback() async {
    screenshotCallback = ScreenshotCallback();

    screenshotCallback.addListener(() {
      print('Take a PRINT');
    });
  }
```

My target SDK is the 29 too, in iOS works fine.

About the permision in storage I have, I can even download files in my app.

**Comment by siloebb** [comment]
I fix it updating the `permission_handler` to `5.1.0+2` version!
I was using the `5.0.1+1`.

**Comment by ousvat** [comment]
the solution is here https://stackoverflow.com/a/64738527/10385430

**Comment by ousvat** [comment]
You can also use this package from my repo where I did more fixes:
```
screenshot_callback:
    git:
      url: git://github.com/ousvat/flutter_screenshot_callback.git
 ```

**Comment by beygee** [comment]
Fixed 2.0.0.
Sorry for being late.
